src/replication.c

```diff
@@ -720,15 +782,37 @@ replica_receiver(void *arg) {
 replica_t *
 create_replica_entry(spec_t *spec, int mgmt_fd)
 {
-	replica_t *replica = (replica_t *)malloc(sizeof(replica_t));
+	replica_t *replica = NULL;
+	int rc;
+
+	replica = (replica_t *)malloc(sizeof(replica_t));
+	if (!replica)
+		return NULL;
+
 	replica->mgmt_fd = mgmt_fd;
-	replica->mgmt_io_state = READ_IO_RESP_HDR;
-	replica->mgmt_io_read = 0;
-	replica->mgmt_ack = (zvol_io_hdr_t *)malloc(sizeof(zvol_io_hdr_t));
-	replica->mgmt_ack_data = (mgmt_ack_data_t *)malloc(sizeof(mgmt_ack_data_t));
+	TAILQ_INIT(&(replica->mgmt_cmd_queue));
 	MTX_LOCK(&spec->rq_mtx);
 	TAILQ_INSERT_TAIL(&spec->rwaitq, replica, r_waitnext);
 	MTX_UNLOCK(&spec->rq_mtx);
+	/*
+	 * As of now, we are supporting single spec_t per target
```

Comment:
this comment is not required here.. its required in accept_mgmt_conns which passes the spec..